### PR TITLE
Remove ListT/ErrorT-related stuff

### PR DESCRIPTION
**Pull request by fumieval**
They are removed from transformers; we should remove them from mtl too (#75)

### CODE REVIEW

**Comment by chessai** [comment]
@fumieval the CPP i was referring to in #75 was for allowing for these instances with older transformers. But, I'm tempted to say that these can just be removed in the next majour version bump, without the need for CPP.

File: cabal.project [hunk 1]
```diff
@@ -1 +1,2 @@
 packages: .
+  transformers/
```
Comment (chessai) [inline cabal.project:2]:
```suggestion
\n
```

File: cabal.project [hunk 2]
```diff
@@ -1 +1,2 @@
 packages: .
+\n
```
Comment (chessai) [inline cabal.project:2]:
```suggestion
```

**Comment by chessai** [comment]
I am confused as to why github is showing that "All checks have failed", because all of them succeeded, except for HEAD. Perhaps it is to do with the strange HEAD failure, which maybe @RyanGlScott can make light of: https://travis-ci.org/github/haskell/mtl/jobs/707484624

**Comment by RyanGlScott** [comment]
I'm not sure what to make of HEAD-related failures either. But for what it's worth, the version of GHC HEAD that is provided at https://launchpad.net/~hvr/+archive/ubuntu/ghc (which `haskell-ci` uses) is quite old (around GHC 8.7 or so),. I've opted to remove the `ghc-head` configurations in all of my `haskell-ci` configs, since a HEAD that old is basically useless.

**Comment by chessai** [comment]
@RyanGlScott thanks, I'll probably just remove it then.

**Comment by chessai** [comment]
Can't get CI for the old HEAD to pass, so I'm going to just disregard it. HEAD has been removed from the .travis.yml on master.

**Pull request merged**